Set+Concentration/Set+Concentration/Views/Set/SetViewController.swift

```diff
@@ -0,0 +1,292 @@
+        didSet {
+            let swipeDownForExtraCards = UISwipeGestureRecognizer(target: self, action: #selector(dealCards(_:)))
+            swipeDownForExtraCards.direction = .down
+            cardDeckView.addGestureRecognizer(swipeDownForExtraCards)
+            cardDeckView.addGestureRecognizer(UIRotationGestureRecognizer(target: self, action: #selector(shuffleCards)))
+        }
+    }
+    
+    //MARK: - Private Properties
+    
+    private var game = Game()
+    private var hintCardViews = [CardView]()
+    private var dealtCardViews = [CardView]()
+    private lazy var animator = UIDynamicAnimator(referenceView: cardDeckView)
+    private lazy var cardBehaviour = CardViewBehaviour(in: animator)
+    private enum UpdateViewCase { case initView, touchCard, giveHint, shuffle, deal3 }
+    private var cardViews = [CardView]() { didSet { cardViews.enumerated().forEach({$0.element.tag = $0.offset})}}
+    private var selectedCardViews: [CardView] {
+        var selectedViews = [CardView]()
+        for card in game.cardsSelected {
+            if let index = game.cardsOnTable.index(of: card) {
+                selectedViews.append(cardViews[index])
+            }
+        }
+        
+        for card in game.cardsOnTable {
+            if game.cardsSelected.contains(card) {
+                if let index = game.cardsOnTable.index(of: card) {
+                    cardViews[index].setCardViewState(state: .selected)
+                }
+            } else {
+                if let index = game.cardsOnTable.index(of: card) {
+                    cardViews[index].setCardViewState(state: .normal)
+                }
+            }
+        }
+        return selectedViews
+    }
+    private var removableCardViews: [CardView] {
+        var removableViews = [CardView]()
+        for card in game.cardsToRemove {
+            if let index = game.cardsOnTable.index(of: card) {
+                removableViews.append(cardViews[index])
+            }
+        }
+        return removableViews
+    }
+    
+    //MARK: - Button Actions
+    
+    @IBAction private func newGame(_ sender: UIButton) {
+        game.reset()
+        dealCardsButton.isEnabled = true
+        iosLabel.alpha = 0.0
+        updateViewFromModel(inCase: .initView)
+    }
+    
+    @IBAction private func dealCards(_ sender: UIButton) {
+        if game.deckCount == 0 {
+            dealCardsButton.isEnabled = false
+        } else {
+            game.dealThreeCards()
+            updateViewFromModel(inCase: .deal3)
+        }
+    }
+    
+    @IBAction func giveHint(_ sender: UIButton) {
+        game.giveHint()
+        hintCardViews.removeAll()
+        for (index, card) in game.cardsOnTable.enumerated() {
+            if game.cardsHint.contains(card) {
+                hintCardViews.append(cardViews[index])
+            }
+        }
+        updateViewFromModel(inCase: .giveHint)
+    }
+    
+    //MARK: - Overriden Methods
+    
+    override func viewDidLoad() {
```

Comment:
methods of the UIViewController life cycle should be in the beginning of the file, before IBActions